### PR TITLE
fix: fallback to now if date is not valid when generating weeks array

### PR DESCRIPTION
**Pull request by jsprds**
https://github.com/vlio20/angular-datepicker/issues/561

### CODE REVIEW

File: src/lib/day-calendar/day-calendar.service.ts [hunk 1]
```diff
@@ -51,25 +51,26 @@ export class DayCalendarService {
   }
 
   generateMonthArray(config: IDayCalendarConfigInternal, month: Moment, selected: Moment[]): IDay[][] {
+    const parsedMonth = month.isValid() ? month.clone() : moment();
```
Comment (vlio20) [inline src/lib/day-calendar/day-calendar.service.ts:54]: Can we cover it with a test? e2e or unit

Comment (jsprds) [reply src/lib/day-calendar/day-calendar.service.ts:54]: Added a unit test ;-)

**Comment by vlio20** [comment]
@jsprds notice that CI is failing

**Comment by jsprds** [comment]
I ran the tests locally, everything is fine. The failing ci pipeline is, like you say, unrelated to my modifications.

I don't have an immediate solution to this problem, I'm not familiar with the setup you're currently using for the e2e testing. 😕 

Maybe you can create a new bug ticket for the issue so it doesn't block this bugfix? It would be nice if this fix could be implemented without the CI fix. 🙏

**Comment by vlio20** [comment]
Yea, I will take a look at it, I would like to run your PR after I will fix the e2e. I hope that I will get to it today or tomorrow,

**Pull request merged**